msfits/MSFits/FitsIDItoMS.cc

```diff
@@ -2062,8 +2059,10 @@ void FITSIDItoMS1::fillMSMainTable(const String& MSFileName, Int& nField, Int& n
 //	    cout<<"    visReal="<< visReal << "visImag="<< visImag<<endl;
 //	  }
 
-	  if(uv_data_hasWeights_p){
-	      memcpy(&visWeight, (static_cast<Float *>(data_addr[iFlux])) + count++, sizeof(Float));     
+	  if (uv_data_hasWeights_p) {
+	    memcpy(&visWeight, (static_cast<Float *>(data_addr[iFlux])) + count++, sizeof(Float));
+	  } else {
```

Comment:
The code in this `else` branch was not there before this change, i.e. if `uv_data_hasWeights_p==False` (data has no weights?). This affects the `count`. Does the code still work for data without weights?